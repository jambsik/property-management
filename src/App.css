html,
body,
#root {
    width: 100%;
    height: calc(100vh + 100px);
    margin: 0;
    padding: 0;
    overflow-x: hidden;
    overflow-y: auto;
}

a {
    text-decoration: none;
}
