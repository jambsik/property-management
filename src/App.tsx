import React from 'react';
import { BrowserRouter, RouteProps, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';

import { appRoutes } from './config/routes';
import { appTheme } from './theme/theme';
import './App.css';

const App: React.FC = () => (
    <ThemeProvider theme={appTheme}>
        <BrowserRouter>
            <Routes>
                {appRoutes.map((props: RouteProps, index: number) => (
                    <Route key={`${props.path || index}`} {...props} />
                ))}
            </Routes>
        </BrowserRouter>
    </ThemeProvider>
);

export default App;
