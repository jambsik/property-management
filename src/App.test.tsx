import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('App container test', () => {
    it('Should render component App with out errors', () => {
        const { container } = render(<App />);

        expect(container).toMatchSnapshot();
    });

    it('Should show home page after init routing in App container', () => {
        const { getByText } = render(<App />);

        expect(getByText('home.title')).toBeInTheDocument();
    });
});
