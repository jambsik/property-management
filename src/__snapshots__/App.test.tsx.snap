// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container test Should render component App with out errors 1`] = `
<div>
  <div
    class="MuiBox-root css-srv7rf"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1ad5ige-MuiTypography-root"
    >
      home.title
    </h1>
    <a
      href="/mock-view"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1mrv2h2-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        home.get_started
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</div>
`;
